### ISSUE TITLE
fromPromise test bug

### ISSUE THREAD

**Issue by paulpdaniels** [Reactive-Extensions/RxJS#976]
Not a huge bug since all the tests are green as far as I can tell, but with the most recent version I noticed that the new scheduler argument removes the ability to isolate test time properly with a `TestScheduler` when using `fromPromise`.

i.e.

```javascript
var hot = scheduler.createHotObservable(onNext(210, 0));
var resolved = scheduler.createResolvedPromise(220, 1);

var result = scheduler.startScheduler(function() {
  return hot
    .flatMap(function(x) {
      return resolved;
    });
});

//Won't work because flatMap will wrap the Promise using the `DefaultScheduler`
result.messages.assertEqual(onNext(220, 1)); 
```

We can still test using an `async()` or always pass in `Promises` as `fromPromise(promise, testScheduler)`, but it seems to kind of defeat the purpose.

**Comment by mattpodwysocki** [comment]
@paulpdaniels yeah, it's an unfortunate tradeoff we had to make in order to escape the Promise jail, so for now that's the only way I can think of in your tests to get around it is to explicitly calls `fromPromise` yourself. Suggestions are welcome though!

**Comment by amoerie** [comment]
An unfortunate tradeoff? So this won't be fixed?

**Comment by mattpodwysocki** [comment]
@amoerie I said it was an unfortunate tradeoff for what I did there. I didn't say it won't be fixed, I'm just looking for a better design which I have yet to find.

**Comment by amoerie** [comment]
Fair enough, thanks for clarifying!
